src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1378,9 +1378,12 @@ public static bool IsAtStartOfPattern(this SyntaxTree syntaxTree, SyntaxToken le
 
             // e is 1 and $$
             // e is 1 or $$
-            if (leftToken.IsKind(SyntaxKind.AndKeyword) || leftToken.IsKind(SyntaxKind.OrKeyword))
+            // e is SomeEnum.SomeEnumValue and $$
+            // e is SomeEnum.SomeEnumValue or $$
+            // 'and' & 'or' are identifier in the last 2 examples because of lack of context
+            if (leftToken.IsKindOrHasMatchingText(SyntaxKind.AndKeyword) || leftToken.IsKindOrHasMatchingText(SyntaxKind.OrKeyword))
```

Comment:
❓ Shouldn't this include `leftToken.IsKindOrHasMatchingText(SyntaxKind.NotKeyword)`? Patterns are not not allowed to use this.

Reply:
Can you please give an example where `not not` is a valid pattern. I cannot understand how it is possible

Reply:
@sharwell `not` is always valid after `not`. This code is valid:
```c#
void M(int a)
{
    if (a is not not 7)
    {
    }
}
```

Reply:
@Neme12 Didn't know that. However, in our case second not will not form a valid pattern:
```cs
var a = DayOfWeek.Monday;

if (a is DayOfWeek.Saturday not not DayOfWeek.Monday)
{

}
```
This code produces 9 errors.